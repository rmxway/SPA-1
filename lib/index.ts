export * from './jest-template';
export * from './template';
