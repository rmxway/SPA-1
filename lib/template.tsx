'use client';

import { memo, ReactNode } from 'react';
import { ThemeProvider } from 'styled-components';

import { Navbar } from '@/components';
import { TopBlock } from '@/components/ui';
import { checkVersion } from '@/services/localStore';
import { store } from '@/store';
import { ReduxProvider } from '@/store/provider';
import { defaultTheme, GlobalStyles } from '@/theme';

export type TemplateProps = {
	children?: ReactNode | undefined;
};

checkVersion(store);

export const Template = memo(({ children }: TemplateProps) => (
	<ReduxProvider>
		<ThemeProvider theme={defaultTheme}>
			<GlobalStyles />
			<Navbar />
			<TopBlock />
			{children}
		</ThemeProvider>
	</ReduxProvider>
));

export default Template;
