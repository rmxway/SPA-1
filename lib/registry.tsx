'use client';

import { useServerInsertedHTML } from 'next/navigation';
import React, { useState } from 'react';
import { ServerStyleSheet, StyleSheetManager } from 'styled-components';

import { JestTemplate, Template } from '@/lib';

interface StyledRegistryProps {
	isJest?: boolean;
	children: React.ReactNode;
}

export function StyledComponentsRegistry({ isJest, children }: StyledRegistryProps) {
	const [sheet] = useState(() => new ServerStyleSheet());

	const Tag = isJest ? JestTemplate : Template;

	try {
		useServerInsertedHTML(() => {
			const styles = sheet.getStyleElement();
			sheet.instance.clearTag();

			// eslint-disable-next-line react/jsx-no-useless-fragment
			return <>{styles}</>;
		});

		if (typeof window !== 'undefined') return <Tag>{children}</Tag>;

		return (
			<StyleSheetManager sheet={sheet.instance}>
				<Tag>{children}</Tag>
			</StyleSheetManager>
		);
	} catch (error) {
		throw new Error((error as Error).message);
	} finally {
		sheet.instance.clearTag();
	}
}

export default StyledComponentsRegistry;
