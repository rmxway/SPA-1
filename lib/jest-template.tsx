import { ThemeProvider } from 'styled-components';

import { ReduxProvider } from '@/store/provider';
import { defaultTheme, GlobalStyles } from '@/theme';

import { TemplateProps } from './template';

export const JestTemplate = ({ children }: TemplateProps) => (
	<ReduxProvider>
		<ThemeProvider theme={defaultTheme}>
			<GlobalStyles />
			{children}
		</ThemeProvider>
	</ReduxProvider>
);

export default JestTemplate;
