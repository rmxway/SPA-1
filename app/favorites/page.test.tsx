import { render } from '@testing-library/react';

import { StyledComponentsRegistry } from '@/lib/registry';

import Page from './page';

describe('Favorite page test:', () => {
	it('Render text', () => {
		const { container, getByText } = render(
			<StyledComponentsRegistry isJest>
				<Page />
			</StyledComponentsRegistry>,
		);

		const text = getByText(/Delete favorites|Nothing was't add to favorites, go to/i);
		expect(text).toBeInTheDocument();
		expect(container).toMatchSnapshot();
	});
});
