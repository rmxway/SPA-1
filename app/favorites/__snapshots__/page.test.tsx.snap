// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorite page test: Render text 1`] = `
<div>
  <div
    class="sc-cda6c8d6-0 fYhgvr"
  >
    <div
      class="sc-4a960974-0 ifLSlP"
    >
      <div
        class="sc-4a960974-1 dwEpXO"
        style="opacity: 0;"
      />
      <div
        class="sc-59cb1f7b-1 gkTKt"
        style="opacity: 0;"
      >
        <span
          class="sc-a1f6370b-0 goVLkk"
        />
      </div>
      <div
        style="opacity: 0;"
      >
        <div
          class="sc-9bb5e335-0 dbqDQw"
        >
          Nothing was't add to favorites, go to
           
          <a
            href="/products"
          >
            Products
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
